### ISSUE TITLE
Use non-privileged user for ssh_push client registration

### ISSUE THREAD

**Issue by agayon** [uyuni-project/uyuni#2793]
I want to use non-root user for ssh-push server registration and patch management. However, I can’t find any documentation regarding this.

This link [contact-methods-pushssh](https://www.uyuni-project.org/uyuni-docs/uyuni/client-configuration/contact-methods-pushssh.html) only seems to apply for bootstrapping through ssh.

From what I can see, this unprivileged user would need write/read permissions on `/etc/yum.repo.d`, `/etc/apt/sources.list.d` or `/etc/zypp` depending of the OS, and permissions to execute yum/apt/zypper commands. And I guess some extra configurations needed as the Uyuni server will sudo the commands.

**Comment by mcalmer** [comment]
With our used configuration management system you can change everything on the target system. This also means that you need to have access to everything. This is why we did not spend much time on using a none root user.
As ubuntu require this, we will look into this more deeply in future, but we always test with full privileges.
Reducing them can result in any kind of error and we cannot test all the cases. It would be up to the admin to find a special crafted sudoers config which fullfill his needs.

**Comment by phibid** [comment]
Yes ok, this makes sense. However we would rather like to use a dedicated uyuni management user with full sudo privileges than root.
This for several reasons, the main one being that most of the audit scan (OSCAP, …) don’t like to have the PermitRootLogin ssh parameter set to anything other than a “no”.

And you will certainly be agree with it, but me and most of the DevOps guys don’t like to allow direct root access on production servers,
even with public key auth restricted (‘without-password’). A ‘last’ command output with lots of ‘root’  lines is just so ….. disturbing 😊

Philippe.


Philippe Bidault |  Unix Engineer |  Getronics
________________________________
M. 34617301667 |  E. Philippe.Bidault@Getronics.com |  W. www.getronics.com<https://www.getronics.com>



Getronics CMC Service Desk Iberia S.L -  VAT No:S.L.: B66686262.
Registered Office - Getronics CMC Service Desk Iberia S.L, C/Rosselloi, Porcel, 21 planta 11, 08016 Barcelona, Spain.

The information transmitted is intended only for use by the addressee and may contain confidential and/or privileged material. Any review, re-transmission, dissemination or other use of it, or the taking of any action in reliance upon this information by persons and/or entities other than the intended recipient is prohibited. If you received this in error, please inform the sender and/or addressee immediately and delete the material.  Thank you.
Legal disclaimer: http://www.getronics.com/legal/
